File: dart/glue/lib/src/lib/bool/is_empty.dart
import 'package:glue/src/eval.dart';
import 'package:glue/src/ir.dart';

final Ir isEmpty_ = IrNativeFunc(isEmptyImpl);

Eval<Ir> isEmptyImpl(Ir arg) => Eval.pure(IrBool(isEmpty(arg)));

File: haskell/glue/src/Glue/Lib/Bool/IsEmpty.hs
module Glue.Lib.Bool.IsEmpty where

import Glue.Eval (Eval)
import Glue.IR (IR (..), isEmpty)

isEmpty_ :: IR Eval
isEmpty_ = NativeFunc isEmptyImpl

isEmptyImpl :: IR Eval -> Eval (IR Eval)
isEmptyImpl = pure . Bool . isEmpty
